src/machine.rs

```diff
@@ -74,7 +74,11 @@ pub struct MemoryExtra {
     pub stacked_borrows: Option<stacked_borrows::MemoryExtra>,
     pub intptrcast: intptrcast::MemoryExtra,
 
+    /// Mapping extern static names to their canonical allocation.
+    pub(crate) extern_statics: HashMap<&'static str, AllocId>,
```

Comment:
I wonder if this should be keyed on `Symbol` instead.
Also, does miri not use `FxHashMap`s? They're an easy speed boost for rustc but idk how relevant that is here.

(Sorry if this stuff has been discussed before in the context of miri, I haven't personally seen anything about this)

Reply:
Miri doesn't currently use `FxHashMap` but probably should, yeah.

As for `Symbol`, I honestly doubt it's worth all the trouble that brings with it.

Reply:
> worth all the trouble that brings with it

You always start with `Symbol` for looking up things and then call `.as_str()` on it, the only thing that's more complex is inserting (you have to call `Symbol::intern("...")`) but that happens only once.